### DISCUSSION TITLE
Dynamically configuring OvenMediaEngine

### DISCUSSION THREAD

**Discussion by deekshithkv** [OvenMediaLabs/OvenMediaEngine#1347] [Help]
Hi, 
Currently we are using OvenMediaEngine to stream HLS/WebRTC with input as RTSP. I am using Server.xml file to configure RTSP input and HLS/WebRTC output streams. Is there any way to configure OvenMediaEngine dynamically while the server is running (without server restart)? Is there any SDK, API to do the same?

Regards

**Comment by getroot** [comment]
OvenMediaEngine supports REST-API, please refer this, 

https://airensoft.gitbook.io/ovenmediaengine/rest-api

**Reply by deekshithkv** [reply]
@getroot Does This API has provision to update the input, output streams. Say if i want to change the input stream URL or output URL will it be possible with this REST API?

**Reply by getroot** [reply]
I don't understand what you mean by changing the output URL, but pulling RTSP can be done in several ways.

Pull Stream while creating a stream with the API
https://airensoft.gitbook.io/ovenmediaengine/rest-api/v1/virtualhost/application/stream#create-stream-pull

Pull Stream using OriginMapStore and REDIS Server
https://airensoft.gitbook.io/ovenmediaengine/live-source/rtsp-pull-beta#pulling-streams-using-the-originmapstore

**Reply by krisdahl** [reply]
Maybe look at the Admission Webhooks? I think that may be what you are looking for. You can redirect incoming and outgoing streams.
